smartsim/_core/_install/buildenv.py

```diff
@@ -539,20 +557,26 @@ def check_build_dependency(self, command: str) -> None:
         except OSError:
             raise SetupError(f"{command} must be installed to build SmartSim") from None
 
-    @staticmethod
-    def check_installed(package: str, version: t.Optional[Version_] = None) -> bool:
+    @classmethod
+    def check_installed(cls, package: str, version: t.Optional[str] = None) -> bool:
         """Check if a package is installed. If version is provided, check if
         it's a compatible version. (major and minor the same)"""
         try:
-            installed = Version_(pkg_resources.get_distribution(package).version)
-            if version:
-                # detect if major or minor versions differ
-                if installed.major != version.major or installed.minor != version.minor:
-                    msg = (
-                        f"Incompatible version for {package} detected.\n"
-                        + f"{package} {version} requested but {package} {installed} installed."
-                    )
-                    raise SetupError(msg)
-            return True
-        except pkg_resources.DistributionNotFound:
+            installed = cls.get_package_version(package)
+        except importlib.metadata.PackageNotFoundError:
             return False
+        if version:
+            if not isinstance(version, Version_):
+                version = Version_(version)
+            # detect if major or minor versions differ
+            if installed.major != version.major or installed.minor != version.minor:
```

Comment:
Could we create a comparison operator on Version_ to make this accessible in other places?

Reply:
We actually do have the comparison operators for `Version_` implemented which just differ to the `packaging.Version` comparison ops. The problem is that those operations compare patch as well as the major and minor which we do not want.

Totally willing to revisit changing this behavior, but I would argue that it is out of scope for this ticket. Thoughts on putting up a new ticket instead?

Reply:
I concur. Good on ya for refusing my feature creep. Maybe a modulo operator feels nice? `if installed % version is None: ...`